firebase-firestore/src/main/java/com/google/firebase/firestore/auth/FirebaseAuthCredentialsProvider.java

```diff
@@ -85,19 +88,21 @@ public FirebaseAuthCredentialsProvider(InternalAuthProvider authProvider) {
     // Take note of the current value of the tokenCounter so that this method can fail (with a
     // FirebaseFirestoreException) if there is a token change while the request is outstanding.
     final int savedCounter = tokenCounter;
-    return res.continueWith(
+    return res.continueWithTask(
+        Executors.DIRECT_EXECUTOR,
         task -> {
           synchronized (this) {
             // Cancel the request since the token changed while the request was outstanding so the
             // response is potentially for a previous user (which user, we can't be sure).
             if (savedCounter != tokenCounter) {
-              throw new FirebaseFirestoreException(
-                  "getToken aborted due to token change", Code.ABORTED);
+              Logger.debug(LOG_TAG, "getToken aborted due to token change");
+              return getToken();
```

Comment:
General observation: it seems a little unfortunate that `getToken` and `getResult().getToken()` look so similar while having a very different effect. In general, I'd expect a `get`-prefixed function to be a simple getter, so it might be better to rename `getToken` to something like `fetchToken` to better convey its semantics. Of course, this is absolutely out of scope for this PR.

Reply:
Ack :)